File: llm_outputs/python_programs_corrected_by_llama/sieve.py
def sieve(max):
    primes = [True] * (max + 1)
    primes[0] = primes[1] = False
    for n in range(2, int(max ** 0.5) + 1):
        if primes[n]:
            for i in range(n * n, max + 1, n):
                primes[i] = False
    return [n for n in range(2, max + 1) if primes[n]]

File: llm_outputs/java_programs_corrected_by_llama/SIEVE.java
package java_programs;

import java.util.ArrayList;

public class Sieve {

    public static boolean all(ArrayList<Boolean> arr) {
        for (Boolean value : arr) {
            if (!value) { 
                return false; 
            }
        }  
        return true;
    }

    public static boolean any(ArrayList<Boolean> arr) {
        for (Boolean value: arr) {
            if (value) { 
                return true; 
            }
        }  
        return false;
    }

    public static ArrayList<Boolean> listComp(int n, ArrayList<Integer> primes) {
        ArrayList<Boolean> builtComprehension = new ArrayList<Boolean>();
        for (Integer p : primes) {
            builtComprehension.add((n % p == 0));
        }
        return builtComprehension;
    }

    public static ArrayList<Integer> sieve(int max) {
        ArrayList<Integer> primes = new ArrayList<>();
        for (int n=2; n<=max; n++) {
            if (all(listComp(n, primes)) ||!listComp(n, primes).isEmpty()) {
                continue;
            }
            primes.add(n);
        }
        return primes;
    }
}